R/vectorized_search_per_species.R

```diff
@@ -0,0 +1,27 @@
+vectorized_search_per_species <- function(parsed_partition, full_genes){
+  gene_names <- parsed_partition$gene_Name
+  #First create a matrix to store if a species has a gene
+  mat1 <- matrix(0,nrow=nrow(full_genes),ncol=length(gene_names),byrow=TRUE)
+  colnames(mat1) <- gene_names
+  rownames(mat1) <- rownames(full_genes)
+  
+  length_gene_places = nrow(parsed_partition)
+  
+  row_number <- nrow(full_genes)
```

Comment:
Remove these and instead do this in the loop initialization instead (i.e., `1:nrow(parsed_partition)`